### ISSUE TITLE
CONFIG_BT_CTLR_OPTIMIZE_FOR_SPEED not selected by default in zephyr 3.7

### ISSUE THREAD

**Issue by marcowidmer** [zephyrproject-rtos/zephyr#76427]
**Describe the bug**
Building a bluetooth application for the NRF52 in zephyr 3.6 automatically selected `CONFIG_BT_CTLR_OPTIMIZE_FOR_SPEED=y`. However, in zephyr 3.7, this default is not selected anymore.

**To Reproduce**
Run the following command in zephyr 3.6 and zephyr 3.7:
```
west build samples/bluetooth/peripheral -p -b nrf52840dongle_nrf52840
grep CONFIG_BT_CTLR_OPTIMIZE_FOR_SPEED build/zephyr/.config
```

**Expected behavior**
`CONFIG_BT_CTLR_OPTIMIZE_FOR_SPEED` should be selected in zephyr 3.7, but it is not.

**Impact**
Bluetooth applications may run slower.

**Logs and console output**

**Environment (please complete the following information):**

**Additional context**
It looks like the reason is that the kconfig `choice` default introduced in #70251 does not apply because it is inside a kconfig `menu` which is hidden through a `visible if`. Enabling the menu fixes the issue:
```
west build samples/bluetooth/peripheral -p -b nrf52840dongle_nrf52840 -- -DCONFIG_BT_CTLR_ADVANCED_FEATURES=y
```

**Comment by marcowidmer** [comment]
@cvinayak

**Comment by cvinayak** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/74916/commits/5736518d82a89f4d7724ba5e6d556307e02be92c

**Comment by marcowidmer** [comment]
Thanks, I wasn't aware of the fix. Looks like it will be merged in #74916
